avaframe/in3Utils/geoTrans.py

```diff
@@ -288,35 +291,85 @@ def remeshDEM(cfg, dem):
-                                       fill_value=headerDEM['noDataValue'])
-        log.info('Remeshed data extent difference x: %f and y %f' % (diffExtentX, diffExtentY))
-        dem['rasterData'] = zNew
-
-    return dem
+        # first check if remeshed DEM is available
+        remeshedDem, DEMFound = searchRemeshedDEM(cfg)
+        if DEMFound:
+            log.debug('remeshed DEM found')
+        else:
+            log.info('Remeshing the input DEM (of cell size %.4g m) to a cell size of %.4g m' % (cszDEM, cszNew))
+            x, y, xNew, yNew, diffExtentX, diffExtentY = getMeshXY(dem, cellSizeNew=cszNew)
+            xGrid, yGrid = np.meshgrid(x, y)
+            xGrid = xGrid.flatten()
+            yGrid = yGrid.flatten()
+            z = dem['rasterData']
+            zCopy = np.copy(z)
+            zCopy = zCopy.flatten()
+            mask = np.where(~np.isnan(zCopy))
+            xGrid = xGrid[mask]
+            yGrid = yGrid[mask]
+            z = zCopy[mask]
+
+            headerRemeshed = {}
+            headerRemeshed['cellsize'] = cszNew
+            headerRemeshed['ncols'] = len(xNew)
+            headerRemeshed['nrows'] = len(yNew)
+            headerRemeshed['xllcenter'] = headerDEM['xllcenter']
+            headerRemeshed['yllcenter'] = headerDEM['yllcenter']
+            headerRemeshed['noDataValue'] = headerDEM['noDataValue']
+
+            remeshedDem = {'header': headerRemeshed}
+            xNewGrid, yNewGrid = np.meshgrid(xNew, yNew)
+            zNew = sp.interpolate.griddata((xGrid, yGrid), z, (xNewGrid, yNewGrid), method='cubic',
+                                           fill_value=headerDEM['noDataValue'])
+            log.info('Remeshed data extent difference x: %f and y %f' % (diffExtentX, diffExtentY))
+            remeshedDem['rasterData'] = zNew
+
+            # save remeshed DEM
+            pathToDem = pathlib.Path(cfg['avalancheDir'], 'Inputs', 'DEMremeshed')
+            fU.makeADir(pathToDem)
+            outFile = pathToDem / ('remeshedDEM%.2f.asc' % remeshedDem['header']['cellsize'])
+            IOf.writeResultToAsc(remeshedDem['header'], remeshedDem['rasterData'], outFile, flip=True)
+
+    return remeshedDem
+
+
+def searchRemeshedDEM(cfg):
+    """ search if remeshed DEM already available
+
+        Parameters
+        -----------
+        cfg: configparser object
+            configuration settings: avaDir, meshCellSize, meshCellSizeThreshold
+
+        Returns
+        --------
+        remshedDEM: dict
+            dictionary of remeshed DEM if not found empty dict
+        DEMFound: bool
+            flag if dem is found
+    """
+
+    # load input parameters
+    cellSize = cfg.getfloat('meshCellSize')
+    cellSizeThreshold = cfg.getfloat('meshCellSizeThreshold')
+
+    # path to remeshed DEM folder
+    pathToDem = pathlib.Path(cfg['avalancheDir'], 'Inputs', 'DEMremeshed')
+    fU.makeADir(pathToDem)
```

Comment:
Why make the folder? You only need to read it no?

Reply:
this directory only exists if before a remeshed DEM has been saved, hence when searching - I changed it to check if dir exists